lib/repo/git_repository.rb

```diff
@@ -391,49 +471,65 @@ def get_users(permissions)
 
-      #if @repos_admin # Are we admin?
-      # Adds a user with given permissions to the repository
+      if @repos_admin # Are we admin?
+        # Adds a user with given permissions to the repository
+        ga_repo = Gitolite::GitoliteAdmin.new(
+          Repository.conf[:REPOSITORY_PERMISSION_FILE])
 
-      ga_repo =
-          Gitolite::GitoliteAdmin.new(Repository.conf[:REPOSITORY_STORAGE])
-      repo = ga_repo.config.get_repo(self.get_repos.workdir.split('/').last)
+        # Sync the admin repo
+        ga_repo.update
+        repo = ga_repo.config.get_repo(self.get_repos.workdir.split('/').last)
 
-      # Gets permissions of a particular user
-      repo.permissions[0].each do |perm|
-        if(repo.permissions[0][perm[0]][""].include? user_id)
-          return self.class.__translate_perms_from_file(perm[0])
+        # Gets permissions of a particular user
+        repo.permissions[0].each do |perm|
+          if(repo.permissions[0][perm[0]][""].include? user_id)
+            return self.class.__translate_perms_from_file(perm[0])
+          end
         end
-      end
-      raise UserNotFound.new(user_id + " not found")
 
-      #else
-      #  raise NotAuthorityError.new("Unable to modify permissions: Not in authoritative mode!")
-      #end
+        raise UserNotFound.new(user_id + " not found")
+
+      else
+        raise NotAuthorityError.new(
+          "Unable to modify permissions: Not in authoritative mode!")
+      end
     end
 
     def set_permissions(user_id, permissions)
       # Set permissions for a single given user
       if @repos_admin # Are we admin?
 
-        #TODO: remove permissions should be done before reseting it
-        # in case he already has a permission
         remove_user(user_id)
 
         # Adds a user with given permissions to the repository
-        ga_repo = Gitolite::GitoliteAdmin.new(Repository.conf[:REPOSITORY_PERMISSION_FILE])
+        ga_repo = Gitolite::GitoliteAdmin.new(
+          Repository.conf[:REPOSITORY_PERMISSION_FILE])
+
+        # Sync gitolite admin repo
+        ga_repo.update
+
+        # Grab the repo from gitolite
         repo_name = self.get_repos.workdir.split('/').last
         repo = ga_repo.config.get_repo(repo_name)
 
         if repo.nil?
+          # Create new repo if it doesn't exist
           repo = Gitolite::Config::Repo.new(repo_name)
         end
 
         git_permission = self.class.__translate_to_git_perms(permissions)
         repo.add_permission(git_permission, "", user_id)
         ga_repo.config.add_repo(repo)
+
+        # Readd the 'git' public key to the gitolite admin repo after changes
+        self.class.readd_admin_key
+
+        # update Gitolite repo
         ga_repo.save_and_apply
+
       else
-        raise NotAuthorityError.new("Unable to modify permissions: Not in authoritative mode!")
+        raise NotAuthorityError.new(
+                  "Unable to modify permissions: Not in authoritative mode!")
```

Comment:
Indent the first parameter one step more than the previous line.<br>Prefer single-quoted strings when you don't need string interpolation or special symbols.